File: androidApp/src/main/java/com/lb/htwktimetable/android/ui/timetable/state/TimetableState.kt
package com.lb.htwktimetable.android.ui.timetable.state

import com.lb.functionalities.data.entities.LectureObjectDto
import com.lb.shared.utils.calendar.*

/**
 * State that gathers displayed timetable data.
 */
data class TimetableState(
    val week: YearWeek? = null,
    val prevWeek: YearWeek? = null,
    val nextWeek: YearWeek? = null,
    val weekDays: List<String> = listOf(
        "Montag",
        "Dienstag",
        "Mittwoch",
        "Donnerstag",
        "Freitag",
        "Samstag",
        "Sonntag"
    ),
    val lectures: List<LectureObjectDto>? = null,
    val today: DateUtil? = null
)

File: iosApp/iosApp/timetable/state/TimetableState.swift
import Foundation
import shared

/*
 State that gathers displayed timetable data.
 */
struct TimetableState {
    var week: YearWeek? = nil
    var prevWeek: YearWeek? = nil
    var nextWeek: YearWeek? = nil
    var weekDays: [String] = [
        "Montag",
        "Dienstag",
        "Mittwoch",
        "Donnerstag",
        "Freitag",
        "Samstag",
        "Sonntag"
    ]
    var lectures: [LectureObjectDto]? = nil
    var today: DateUtil? = nil
}
